cmd/agent/app/flags.go

```diff
@@ -24,54 +24,71 @@ import (
 	"flag"
 	"fmt"
 	"strings"
+	"github.com/spf13/viper"
 )
 
+type stringSliceFlag struct {
+	slice *[]string
+}
+
 // Bind binds the agent builder to command line options
-func (b *Builder) Bind(flags *flag.FlagSet) {
+func (b *Builder) GenerateFlags() *flag.FlagSet {
+	flagSet := &flag.FlagSet{}
+
 	for i := range b.Processors {
 		p := &b.Processors[i]
 		name := "processor." + string(p.Model) + "-" + string(p.Protocol) + "."
-		flags.IntVar(
-			&p.Workers,
+		flagSet.Int(
 			name+"workers",
 			p.Workers,
```

Comment:
I want to all these `"server-max-packet-size"` extract to constants. 

How we move this forward? From now viper could be used to manage configuration files, props and env vars.  See the precedence order: https://github.com/spf13/viper#why-viper .